scout/server/blueprints/institutes/controllers.py

```diff
@@ -628,28 +605,44 @@ def clinvar_lines(clinvar_objects, clinvar_header_obj):
     return clinvar_lines
 
 
-def update_HGNC_symbols(store, variant_genes, genome_build):
-    """Update the HGNC symbols if they are not set
-    Returns:
-        gene_object()"""
+def update_variant_genes(store, variant_obj, genome_build):
+    """Update the HGNC symbols and HGVS predictions if they are not set
 
-    if variant_genes is not None:
-        for gene_obj in variant_genes:
-            # If there is no hgnc id there is nothin we can do
-            if not gene_obj["hgnc_id"]:
-                continue
-            # Else we collect the gene object and check the id
-            if gene_obj.get("hgnc_symbol") is None or gene_obj.get("description") is None:
-                hgnc_gene_caption = store.hgnc_gene_caption(gene_obj["hgnc_id"], build=genome_build)
-                if not hgnc_gene_caption:
-                    continue
-                gene_obj["hgnc_symbol"] = hgnc_gene_caption["hgnc_symbol"]
-                gene_obj["description"] = hgnc_gene_caption["description"]
+    Accepts:
+        store(adapter.MongoAdapter)
+        variant_obj(scout.models.variant.Variant): a dictionary
+        genome_build(str): "37" or "38"
+    """
+
+    hgvs_c = []
+    hgvs_p = []
+    gene_symbols = set()
+
+    for gene_obj in variant_obj.get("genes", []):
+        hgnc_id = gene_obj.get("hgnc_id")
+        if hgnc_id is None:
+            continue
+        gene_caption = store.hgnc_gene_caption(hgnc_id, genome_build)
```

Comment:
This line of code had a bug before, since it was not passing genome_build to the function, so all gene variants were searched among build 37

Reply:
💯